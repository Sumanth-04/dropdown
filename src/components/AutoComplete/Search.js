import React, { Component } from "react";
import Select from "rc-select";
//import AutoComplete from "react-autocomplete";
import PropTypes from "prop-types";
import "./autoComplete.scss";
//import "./index.css";

class Search extends Component {
  static propTypes = {
    options: PropTypes.arrayOf(PropTypes.element),
  };

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="">
        <div className="rd-searchBar">
          <i className="fa fa-search"></i>
          <p>|</p>
          <Select
            className="rd-searchBar-primary"
            options={this.props.options}
            mode="combobox"
            notFoundContent="Enter Make or Model"
            style={{ ...this.props.style }}
            onChange={this.props.onChange}
            {...this.props}
            placeholder="Enter Make or Model"
          />
        </div>
      </div>
    );
  }
}

export default Search;
