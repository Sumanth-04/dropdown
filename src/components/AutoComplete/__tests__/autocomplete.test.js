import React from "react";
import Search from "../Search";
import renderer from 'react-test-renderer';

const options = [<div>HI</div>, <div>HI</div>, <div>HI</div>];

it("Rendering AutoComplete", () => {
  const tree = renderer.create(
    <Search options={options} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
