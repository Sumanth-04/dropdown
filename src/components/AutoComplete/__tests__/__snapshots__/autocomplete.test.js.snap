// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering AutoComplete 1`] = `
<div
  className=""
>
  <div
    className="rd-searchBar"
  >
    <i
      className="fa fa-search"
    />
    <p>
      |
    </p>
    <div
      className="rc-select rd-searchBar-primary rc-select-single rc-select-show-search"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      style={Object {}}
    >
      <div
        className="rc-select-selector"
        onMouseDown={[Function]}
      >
        <span
          className="rc-select-selection-search"
        >
          <input
            aria-activedescendant="undefined_list_0"
            aria-autocomplete="list"
            aria-controls="undefined_list"
            aria-haspopup="listbox"
            aria-owns="undefined_list"
            autoComplete="off"
            className="rc-select-selection-search-input"
            onChange={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onPaste={[Function]}
            readOnly={false}
            role="combobox"
            style={
              Object {
                "opacity": null,
              }
            }
            unselectable={null}
            value=""
          />
        </span>
        <span
          className="rc-select-selection-placeholder"
        >
          Enter Make or Model
        </span>
      </div>
    </div>
  </div>
</div>
`;
